### DISCUSSION TITLE
tasmota32 does not listen to cmnd/%topic%/PowerN

### DISCUSSION THREAD

**Discussion by msetina** [arendst/Tasmota#20142] [General]
I have a button on tasmota32 designated Power3. Full topic is the default one. From HA I noticed that the button is not responding. After a little investigation I found out that OFF and ON is sent to cmnd/heat_storage_K/Power3 . The stat/heat_storage_K/Power3 is used and HA gets the actual state. the cmnd call is not shown in the console of the device.

**Comment by msetina** [comment]
If you write a driver in berry make sure that mqtt_data checks the topic and returns false for all topics you are not interested in. Tasota will send all topics to this function not only those that you subscribed (as of version 13.2.0)

**Reply by msetina** [reply]
This fixed my issue.

**Reply by sfromis** [reply]
In Berry, what I'd recommend is to **not** use `mqtt_data` at all, but instead specify a function on the `mqtt.subscribe`, which will then only get messages for the matching topic, and with no need for returning any true/false flag.

**Reply by msetina** [reply]
is signature still def f(topic, idx, payload_s, payload_b) ? the same as mqtt_data... but without return..
And tasmota manages the subscription after mqtt connect so we do not have to make rules. Yes or no?

**Reply by sfromis** [reply]
Yes to all. You can even do `mqtt.subscribe` straight from `autoexec.be` before Wifi/MQTT connection, and Tasmota will manage, also in case of temporary loss of Wifi.
